flask/app/datasets.py

```diff
@@ -138,3 +138,65 @@ def update_dataset(id: int):
     routes.transaction_or_abort(db.session.commit)
 
     return jsonify(table)
+
+
+@app.route("/api/datasets", methods=["POST"])
+@login_required
+def post_dataset():
+    if not request.json:
+        return "Request body must be JSON", 400
+
+    dataset_type = request.json.get("dataset_type")
+    if not dataset_type:
+        return "A dataset type must be provided", 400
+
+    tissue_sample_id = request.json.get("tissue_sample_id")
+    if not tissue_sample_id:
+        return "A tissue sample id must be provided", 400
+
+    models.TissueSample.query.filter_by(
+        tissue_sample_id=tissue_sample_id
+    ).first_or_404()
+
+    col_names = [key for key in request.json]
+    enum_error = routes.enum_validate(models.Dataset, request.json, col_names)
+
+    if enum_error:
+        return enum_error, 400
+
+    try:
+        created_by = updated_by = current_user.user_id
+    except:  # LOGIN DISABLED
+        created_by = updated_by = 1
+
+    dataset = models.Dataset(
+        **{
+            "tissue_sample_id": tissue_sample_id,
+            "dataset_type": dataset_type,
+            "input_hpf_path": request.json.get("input_hpf_path"),
+            "notes": request.json.get("notes"),
+            "condition": request.json.get("condition"),
+            "extraction_protocol": request.json.get("extraction_protocol"),
+            "capture_kit": request.json.get("capture_kit"),
+            "library_prep_method": request.json.get("library_prep_method"),
+            "library_prep_date": request.json.get("library_prep_date"),
+            "read_length": request.json.get("read_length"),
+            "read_type": request.json.get("read_type"),
+            "sequencing_id": request.json.get("sequencing_id"),
+            "sequencing_date": request.json.get("sequencing_date"),
+            "sequencing_centre": request.json.get("sequencing_centre"),
+            "batch_id": request.json.get("batch_id"),
+            "created_by": created_by,
+            "updated_by": updated_by,
+            "discriminator": request.json.get("discriminator"),
+        }
+    )
+    try:
```

Comment:
minor - you can wrap the try/except statement into a single line using `routes.transaction_or_abort(db.session.commit)`

Reply:
nice, that's clean!